utils/lineinfile/lineinfile.go

```diff
@@ -47,16 +47,23 @@ func NewAbsentParams() Params {
 }
 
 func (l LineInFile) Present(params Params) error {
-	if _, err := os.Stat(l.FilePath); err != nil {
+	createFile := false
+	if _, err := os.Stat(l.FilePath); os.IsNotExist(err) {
+		// will be created by atomic.WriteFile
+		createFile = true
+	} else if err != nil {
 		return err
 	}
 
-	content, err := os.ReadFile(l.FilePath)
-	if err != nil {
-		return err
-	}
+	var lines []string
+	if !createFile {
```

Comment:
It this is only checking if something is present, then creating a file seems a rather surprising side effect :thinking:

Reply:
As I said in the other comment, `Present` ensures the file contains a given line, so the file must always exist in that case.